kafka-admin/src/main/java/org/bf2/admin/kafka/admin/RecordOperations.java

```diff
@@ -120,23 +120,6 @@
     }
 
     public CompletionStage<Types.Record> produceRecord(String topicName, Types.Record input) {
-        String key = input.getKey();
-        List<Header> headers = input.getHeaders() != null ? input.getHeaders()
-            .entrySet()
-            .stream()
-            .map(h -> new Header() {
-                @Override
-                public String key() {
-                    return h.getKey();
-                }
-
-                @Override
-                public byte[] value() {
-                    return h.getValue().getBytes();
-                }
-            })
-            .collect(Collectors.toList()) : Collections.emptyList();
-
         CompletableFuture<Types.Record> promise = new CompletableFuture<>();
         Producer<String, String> producer = clientFactory.createProducer();
```

Comment:
Should the producer move down into `send`?

Reply:
I think that should stay here since it's the client used for the pre-`send` validation of the topic/partitions.